src/components/fields/ArrayField.js

```diff
@@ -463,12 +463,25 @@ class ArrayField extends Component {
       if (this.state.originalFormData) {
         this.props.onChange(this.state.originalFormData);
       } else {
+        const {
+          schema,
+          registry = getDefaultRegistry(),
+          formData
+        } = this.props;
+        const { definitions } = registry;
+
+        let itemSchema = schema.items;
+
+        if (isFixedItems(schema) && allowAdditionalItems(schema)) {
+          itemSchema = schema.additionalItems;
+        }
+
         this.props.onChange(
-          getDefaultFormState(
-            this.props.schema,
-            undefined,
-            this.props.registry.definitions
-          )
+          [
+            ...formData,
+            getDefaultFormState(itemSchema, undefined, definitions)
+          ],
+          { validate: false }
```

Comment:
Did you verified that passing `{ validate: false }`  attribute is correct in this case?